File: Java/229.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        Map<Integer, Integer> frequency = new HashMap<>();
        List<Integer> result = new ArrayList<>();
        int limit = nums.length / 3;
        for (int i = 0; i < nums.length; i++) {
            if (frequency.getOrDefault(nums[i], 0) > limit) {
                continue;
            }
            frequency.put(nums[i], frequency.getOrDefault(nums[i], 0) + 1);
            if (frequency.get(nums[i]) > limit) {
                result.add(nums[i]);
            }
        }
        return result;
    }
}

File: Python/229.py
from collections import defaultdict
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        count = defaultdict(int)

        for num in nums:
            count[num] += 1

            if len(count) <= 2:
                continue

            new_count = defaultdict(int)
            for num, count in count.items():
                if count > 1:
                    new_count[num] = count - 1
            count = new_count

        result = []
        for num in count:
            if nums.count(num) > len(nums) // 3:
                result.append(num)

        return result
